### ISSUE TITLE
Error Code: 101 when attempting to capture from Android 7.0

### ISSUE THREAD

**Issue by weschao** [jgoldshlag/padherder_proxy#47]
I'm able to connect to the proxy - I can see the log lines where it connects to dl-na.padsv.gungho.jp, and I can connect to websites in Chrome through the proxy as well. However, the game doesn't let me get past the title screen, giving an Error Code: 101.

Log messages from Android say:

System.err: javax.net.ssl.SSLHandshakeException: java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.
System.err: Caused by: java.security.cert.CertificateException: java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.

https://developer.android.com/training/articles/security-ssl.html says that this error can be due to a number of possibilities, but it seems likely that the problem is that the certificate is self-signed.

This is on a Samsung S6 running Android 7.0. Interestingly, it doesn't reproduce on a Nexus 5 running Android 6.0.1. I also don't reproduce it on iOS, as long as I turn on "enable full trust" for the mitmproxy certificate.

**Comment by weschao** [comment]
https://android-developers.googleblog.com/2016/07/changes-to-trusted-certificate.html suggests that the problem is that the certificate is a user certificate, which is no longer trusted for apps targeting Android 7.0.

**Comment by jgoldshlag** [comment]
Yeah, I think you are going to be out of luck on 7.0, reading that link. Sorry...

**Comment by weschao** [comment]
Bummer. I think this means that padherder_proxy won't work with any version of Android after 7.0.

I'm exploring the possibility of creating an apk that explicitly trusts user certificates and will update here if that's successful.

**Comment by dongwoo1005** [comment]
Man, I spent hours trying to setup on Mac and got it running after small fixes to errors mentioned in #30 to get it working on Mac. Eventually to find out that I cannot get past on Error 101 and find out this issue a few hours later.

@jgoldshlag I suggest you explicitly write on Readme that it does not work on any Android with version 7.0 +. Also, if the project is active, creating a requirements.txt from `pip freeze` and a install script to install all the dependencies would be a nice addition for anyone who wants to compile to run on Mac/Linux or contribute to the project.

FYI, the lines causing error were `host = config.Read("host") or socket.gethostbyname(socket.gethostname())`. I had to change all of them to `host = config.Read("host") or socket.gethostbyname('')`. I am not creating a PR because it could be a hack to get it working on Mac and I have not tried it on Windows.

**Comment by jgoldshlag** [comment]
I don't use Android, and don't keep up with what works and doesn't work there.

I wouldn't accept a PR with that change, and it seems like an odd thing to have to do. If it wasn't binding to the correct IP, that is why I let you specify an IP in the settings (that is what the "config.read("host")" thing is reading. Rather than change the code, have you tried putting in the correct IP address there?
